CommunityToolkit.Uwp.Graph.Controls/Helpers/RoamingSettings/RoamingSettingsHelper.cs

```diff
@@ -0,0 +1,161 @@
+using CommunityToolkit.Net.Authentication;
+using CommunityToolkit.Net.Graph.Extensions;
+using CommunityToolkit.Uwp.Graph.Common;
+using Microsoft.Toolkit.Uwp.Helpers;
+using Windows.Storage;
+
+namespace CommunityToolkit.Uwp.Graph.Helpers.RoamingSettings
+{
+    /// <summary>
+    /// An enumeration of the available data storage methods for roaming data.
+    /// </summary>
+    public enum RoamingDataStore
+    {
+        /// <summary>
+        /// Store data using open extensions on the Graph User.
+        /// </summary>
+        UserExtensions,
+    }
+
+    /// <summary>
+    /// A helper class for syncing data to roaming data store.
+    /// </summary>
+    public class RoamingSettingsHelper : IRoamingSettingsDataStore
+    {
+        /// <summary>
+        /// Gets the internal data storage helper instance.
+        /// </summary>
+        public IRoamingSettingsDataStore DataStore { get; private set; }
+
+        /// <inheritdoc />
+        public IDictionary<string, object> Settings => DataStore?.Settings;
+
+        /// <summary>
+        /// Creates a new RoamingSettingsHelper instance for the currently signed in user.
+        /// </summary>
+        /// <param name="dataStore">Which specific data store is being used.</param>
+        /// <param name="autoSync">Whether the values should immediately sync or not.</param>
+        /// <param name="serializer">An object serializer for serialization of objects in the data store.</param>
+        /// <returns>A new instance of the RoamingSettingsHelper configured for the current user.</returns>
+        public static async Task<RoamingSettingsHelper> CreateForCurrentUser(RoamingDataStore dataStore = RoamingDataStore.UserExtensions, bool autoSync = true, IObjectSerializer serializer = null)
+        {
+            var provider = ProviderManager.Instance.GlobalProvider;
+            if (provider == null || provider.State != ProviderState.SignedIn)
+            {
+                throw new InvalidOperationException("The GlobalProvider must be set and signed in to create a new RoamingSettingsHelper for the current user.");
+            }
+
+            var me = await provider.Graph().Me.Request().GetAsync();
+            return new RoamingSettingsHelper(me.Id, dataStore, autoSync, serializer);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoamingSettingsHelper"/> class.
+        /// </summary>
+        /// <param name="userId">The id of the target Graph User.</param>
+        /// <param name="dataStore">Which specific data store is being used.</param>
+        /// <param name="autoSync">Whether the values should immediately sync or not.</param>
+        /// <param name="serializer">An object serializer for serialization of objects in the data store.</param>
+        public RoamingSettingsHelper(string userId, RoamingDataStore dataStore = RoamingDataStore.UserExtensions, bool autoSync = true, IObjectSerializer serializer = null)
```

Comment:
`autoSync` might be misunderstood as meaning that values will get synced as they are added to the data store, but it seems like this is only handling syncing when the store is first created. Consider renaming to `syncWhenInitialized` or something similar.

Reply:
It might be worth having both. Since most of the methods are synchronous (side effect of extending IObjectStorageHelper) it's not obvious when the data is actually saved. Autosync could be a nice way to  allow devs to NOT sync until they explicitly want to. Default would be true. 

Separately, I agree that `syncWhenInitialized` (or maybe `syncOnInit`?) is more clear than `autosync` for that purpose.

Reply:
Agree, both make sense, gives the dev more control. `syncOnInit` makes sense too